packages/core/src/utils/security/handlers/bearer.ts

```diff
@@ -0,0 +1,32 @@
+import { fromNullable, getOrElse, map } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get } from 'lodash';
+import { SecurityScheme } from './types';
+import { when } from './utils';
+
+const bearerHandler = (msg: string, someInput: unknown, name: string, resource?: unknown) => {
+  return when(isBearerToken(get(someInput, 'headers')), msg, resource);
+};
+
+function isBearerToken(inputHeaders: Headers) {
+  return pipe(
+    fromNullable(get(inputHeaders, 'authorization')),
+    map(authorization => !!authorization.match(/^Bearer\s.+$/)),
+    getOrElse(() => false),
+  );
+}
+
+export const bearer = {
+  test: ({ type, scheme }: SecurityScheme) => scheme === 'bearer' && type === 'http',
+  handle: bearerHandler.bind(undefined, 'Bearer'),
```

Comment:
what's safer? `(undefined).abc` or `({}).abc`?

Reply:
`bind` with undefined does not mean that the `thisArg` for the function is undefined. 

> If no arguments are provided to bind, the this of the executing scope is treated as the thisArg for the new function.

Basically it will propagate the current this scope instead of creating a new one. It's like doing `handle: () => bearerHandler('Bearer')`